utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImpl.kt

```diff
@@ -70,4 +76,33 @@ class PerformanceMetricsAssessorImpl @Inject constructor(
       else -> HIGH_MEMORY_TIER
     }
   }
+
+  override fun getCurrentCpuUsageParameters(
+    currentApplicationState: ApplicationState
+  ): CpuUsageParameters {
+    return CpuUsageParameters.newBuilder().apply {
+      cpuTime = Process.getElapsedCpuTime()
+      processTime = oppiaClock.getCurrentTimeMs()
+      applicationState = currentApplicationState
+      numberOfActiveCores = getNumberOfCores()
+    }.build()
+  }
+
+  override fun getRelativeCpuUsage(
+    cpuUsageAtStartOfTimeWindow: CpuUsageParameters,
+    cpuUsageAtEndOfTimeWindow: CpuUsageParameters
+  ): Double {
+    val deltaCpuTimeMs = cpuUsageAtEndOfTimeWindow.cpuTime - cpuUsageAtStartOfTimeWindow.cpuTime
+    val deltaProcessTimeMs =
+      (cpuUsageAtEndOfTimeWindow.processTime * cpuUsageAtEndOfTimeWindow.numberOfActiveCores) -
```

Comment:
Nit: the core multiplication can be factored out of the subtraction (which should simplify this a tad).

Reply:
done.